cli/Tests/TuistServerTests/Client/ServerDeleteCredentialsOnUnauthorizedMiddlewareTests.swift

```diff
@@ -0,0 +1,75 @@
+import Foundation
+import HTTPTypes
+import Mockable
+import OpenAPIRuntime
+import Testing
+@testable import TuistServer
+
+struct ServerDeleteCredentialsOnUnauthorizedMiddlewareTests {
+    struct TestError: Error, Equatable {}
+
+    private let subject: ServerDeleteCredentialsOnUnauthorizedMiddleware
+    private let serverAuthenticationController = MockServerAuthenticationControlling()
+
+    init() {
+        subject = ServerDeleteCredentialsOnUnauthorizedMiddleware(serverAuthenticationController: serverAuthenticationController)
+    }
+
+    @Test func deletesCredentials_when_unAuthorizedResponse() async throws {
+        let request = HTTPRequest(method: .get, scheme: "https", authority: nil, path: "/api")
+        let baseURL = URL(string: "https://tuist.dev")!
+        let operationID = UUID().uuidString
+        let response = HTTPResponse(status: .unauthorized)
+        given(serverAuthenticationController).deleteCredentials(serverURL: .value(baseURL)).willReturn()
+
+        let (middlewareResponse, _) = try await subject.intercept(
+            request,
+            body: nil,
+            baseURL: baseURL,
+            operationID: operationID
+        ) { _, body, _ in
+            return (response, body)
+        }
+
+        #expect(middlewareResponse == response)
+    }
+
+    @Test func bubbleCredentialDeletionErrors_when_unAuthorizedResponse_and_deletionFails() async throws {
```

Comment:
is this really something we want to test for? I find such a test a little bit odd. If we did for any call that throws, our test suites would be x times larger just testing this